framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -1527,12 +1697,38 @@ public AnnotationMirror convertArrayLenToArrayLenRange(AnnotationMirror arrayLen
         return createArrayLenRangeAnnotation(Collections.min(values), Collections.max(values));
     }
 
+    /** Converts an {@code @StringVal} annotation to an {@code @ArrayLenRange} annotation. */
+    public AnnotationMirror convertStringValToArrayLenRange(AnnotationMirror stringValAnno) {
+        List<String> values = getStringValues(stringValAnno);
+        List<Integer> lengths = ValueCheckerUtils.getLengthsForStringValues(values);
+        return createArrayLenRangeAnnotation(Collections.min(lengths), Collections.max(lengths));
+    }
+
+    /** Converts an {@code @StringVal} annotation to an {@code @ArrayLen} annotation. */
+    public AnnotationMirror convertStringValToArrayLen(AnnotationMirror stringValAnno) {
+        List<String> values = getStringValues(stringValAnno);
+        return createArrayLenAnnotation(ValueCheckerUtils.getLengthsForStringValues(values));
+    }
+
```

Comment:
I agree that the way they are used right now is fine, but I'm concerned with maintenance. I expect that we will want a unified way to convert an arbitrary `StringVal` to an `ArrayLen` or `ArrayLenRange`, and, more importantly, I don't think that future programmers should be presented with two public methods that have identical arguments, very similar names, but important differences in their preconditions. Maybe a compromise is to make these two methods private, and add a public method that takes a `StringVal` and dispatches to these two methods as appropriate?